src/components/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -43,27 +43,50 @@ export const ConfirmationModal = ({
   onConfirm,
   ...props
 }: Props): ReactElement => {
-  return (
-    <Modal
-      buttonRow={
-        <>
-          {confirmExtra}
-          <Button className="u-no-margin--bottom" onClick={props.close}>
-            {cancelButtonLabel}
-          </Button>
-          <Button
-            appearance={confirmButtonAppearance}
-            className="u-no-margin--bottom"
-            onClick={onConfirm}
-          >
-            {confirmButtonLabel}
-          </Button>
-        </>
+  const modalRef: MutableRefObject<HTMLElement> = useRef(null);
+
+  useEffect(() => {
+    if (modalRef.current) {
+      const positiveButton: HTMLButtonElement | null =
+        modalRef?.current?.querySelector("button.p-button--positive");
+      // If the modal has a positive button then focus on that after opening
+      // to make a better keyboard navigation experience.
+      if (positiveButton) {
+        console.log("positive button focus");
+        positiveButton.focus();
+      } else {
+        // If there is no button then focus on the modal wrapper.
+        modalRef.current.focus();
       }
-      {...props}
+    }
+  }, []);
+
+  return (
+    <div
+      className="p-confirmation-modal"
+      ref={modalRef as RefObject<HTMLDivElement>}
     >
-      {children}
-    </Modal>
+      <Modal
```

Comment:
The additional `div` wrapper with the unused `p-confirmation-modal` class is unnecessary: we can set the ref to the `Modal` directly - which is a `div` itself.